dockstore-webservice/src/main/java/io/dockstore/webservice/resources/UserResource.java

```diff
@@ -871,8 +886,30 @@ public Set<Entry> getStarredWorkflows(@ApiParam(hidden = true) @Parameter(hidden
     public Set<Entry> getStarredServices(@ApiParam(hidden = true) @Parameter(hidden = true, name = "user")@Auth User user) {
         User u = userDAO.findById(user.getId());
         checkNotNullUser(u);
-        return u.getStarredEntries().stream().filter(element -> element instanceof Service)
-                .collect(Collectors.toCollection(LinkedHashSet::new));
+        return this.getStarredEntries(u, EntryType.SERVICE);
+    }
+
+    /**
+     * Helper function for getStarred endpoints
+     *
+     * @param user
+     * @param entryType
+     * @return a user's starredEntries for desired entryType
+     */
+    public Set<Entry> getStarredEntries(User user, EntryType entryType) {
+        if (entryType == EntryType.WORKFLOW) {
+            return user.getStarredEntries().stream().filter(BioWorkflow.class::isInstance)
```

Comment:
Getting closer. but I think may be able to avoid repeating the `user.getStarredEntries().stream()` and `.collect(Collectors.toCollection(LinkedHashSet::new));` bits. 

For example, you can save the method reference in a variable https://stackoverflow.com/questions/29219984/can-i-store-a-method-in-a-variable-in-java-8  and then use it later